compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1972,4 +1972,36 @@ object messages {
     val msg = hl"Cannot extend ${"sealed"} $pclazz in a different source file"
     val explanation = "A sealed class or trait can only be extended in the same file as its declaration"
   }
+
+  case class UnableToEmitSwitch()(implicit ctx: Context)
+  extends Message(UnableToEmitSwitchID) {
+    val kind = "Syntax"
+    val msg = hl"""Could not emit switch for ${"@switch"} annotated match"""
+    val explanation = {
+      val errorCodeExample =
+        """val Middle = 'm'
+          |val Last   = 'z'
+          |
+          |val number   = (Middle: @switch) match {
+          |  case 'a'  => 1
+          |  case 'm'  => 13
+          |  case Last => 26  //a non-literal may prevent switch generation: this would not compile.
+          |  case _    => 0
+          |}""".stripMargin
```

Comment:
A more complete example:
```scala
val ConstantB = 'B'
final val ConstantC = 'C'
def tokenMe(ch: Char) = (ch: @switch) match {
  case '\t' | '\n' => 1
  case 'A'         => 2
  case ConstantB   => 3  // a non-literal may prevent switch generation: this would not compile
  case ConstantC   => 4  // a constant value is allowed
  case _           => 5
}
```

Reply:
Awesome! Updated.